optaplanner-core/src/main/java/org/optaplanner/core/api/solver/Solver.java

```diff
@@ -35,19 +35,50 @@
 import org.optaplanner.core.impl.solver.termination.Termination;
 
 /**
- * A Solver solves a planning problem.
- * Clients usually call {@link #solve} and then {@link #getBestSolution()}.
+ * A Solver solves a planning problem and returns the best solution found.
+ * It's recommended to create a new Solver instance for each dataset.
+ * <p>
+ * To create a Solver, use {@link SolverFactory#buildSolver()}.
+ * To solve a planning problem, call {@link #solve(Object)}.
+ * To solve a planning problem without blocking the current thread, use {@link SolverManager} instead.
  * <p>
  * These methods are not thread-safe and should be called from the same thread,
  * except for the methods that are explicitly marked as thread-safe.
  * Note that despite that {@link #solve} is not thread-safe for clients of this class,
  * that method is free to do multithreading inside itself.
- * <p>
- * Build by a {@link SolverFactory}.
  * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
  */
 public interface Solver<Solution_> {
 
+    /**
+     * Solves the planning problem and returns the best solution encountered
+     * (which might or might not be optimal, feasible or even initialized).
+     * <p>
+     * It can take seconds, minutes, even hours or days before this method returns,
+     * depending on the {@link Termination} configuration.
+     * To terminate a {@link Solver} early, call {@link #terminateEarly()}.
+     * @param problem never null, a {@link PlanningSolution}, usually its planning variables are uninitialized
+     * @return never null, but it can return the original, uninitialized {@link PlanningSolution} with a {@link Score} null.
```

Comment:
I think this should say "uninitialized PlanningSolution with a null Score" instead of "uninitialized PlanningSolution with a Score null".

Reply:
Yes, fixing